clustering/status.go

```diff
@@ -258,6 +260,45 @@ func (p *managerProcess) GatherStatus(ctx context.Context) (*StatusSet, error) {
 	return ss, nil
 }
 
+// containErrantTransactions check whether a GTID set contains errant transactions.
+// When the primary load is high, in the rare case, gtid_executed of replicas precedes the primary. (Probably only one event)
+// Assuming such a situation, this function ignores primary's event.
+func containErrantTransactions(primaryUUID, gtidSet string) bool {
+	if primaryUUID == "" {
+		panic("uuid must be set")
+	}
+	if gtidSet == "" {
+		return false
+	}
+
+	// Syntax of GTID set
+	//   gtid_set:
+	//       uuid_set [, uuid_set] ...
+	//       | ''
+	//   uuid_set:
+	//       uuid:interval[:interval]...
+	//   uuid:
+	//       hhhhhhhh-hhhh-hhhh-hhhh-hhhhhhhhhhhh
+	//   h:
+	//       [0-9|A-F]
+	//   interval:
+	//       n[-n]
+	//   (n >= 1)
+	// ref: https: //dev.mysql.com/doc/refman/8.0/en/replication-gtids-concepts.html
+
+	uuidSets := strings.Split(gtidSet, ",")
+	if len(uuidSets) > 1 {
+		return true
+	}
+	split := strings.Split(uuidSets[0], ":")
+	if split[0] != primaryUUID {
+		return true
+	}
+	// If two or more events exist, they are considered errant transactions.
```

Comment:
I'm a bit concerned about this because MySQL has a feature called "group commit".
https://dev.mysql.com/worklog/task/?id=5223

With the group commit, multiple transactions can be committed at once.

Reply:
Thank you!
As you pointed out, there is a possibility that multiple events will be executed at once by "group commit".
So I will allow multiple events as well.

72e68ca